src/argonms/common/loading/item/NXItemDataLoader.java

```diff
@@ -0,0 +1,413 @@
+package argonms.common.loading.item;
+
+import argonms.common.StatEffect;
+import argonms.common.character.inventory.InventoryTools;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import us.aaronweiss.pkgnx.EagerNXFile;
+import us.aaronweiss.pkgnx.NXFile;
+import us.aaronweiss.pkgnx.NXNode;
+
+public class NXItemDataLoader extends ItemDataLoader {
+
+    private final String dataPath;
+
+    protected NXItemDataLoader(String wzPath) {
+        this.dataPath = wzPath;
+    }
+
+    @Override
+    public boolean loadAll() {
+        try {
+            NXFile file = new EagerNXFile(dataPath + "Item.nx" + File.separatorChar);
+            for (NXNode baseNode : file.getRoot()) {
+                if (baseNode.getName().equals("Pet")) {
+                    for (NXNode imgNode : baseNode) {
+                        int itemId = Integer.parseInt(imgNode.getName().replace(".img", ""));
+                        doWork(itemId, imgNode);
+                        loaded.add(itemId);
+                    }
+                } else {
+                    for (NXNode imgNode : baseNode) {
+                        for (NXNode node : imgNode) {
+                            int itemId = Integer.parseInt(node.getName()); // Does not end with .img
+                            doWork(itemId, node);
+                            loaded.add(itemId);
+                        }
+                    }
+                }
+            }
+            file = new EagerNXFile(dataPath + "Character.nx" + File.separatorChar);
+            for (NXNode baseNode : file.getRoot()) {
+                String name = baseNode.getName();
+
+                // .img and Afterimage nodes in the root are excluded as we don't use them.
+                // Face and Hair nodes in the root are excluded as we already load them in NXBeautyDataLoader.
+                if (!name.endsWith(".img") && !name.equals("Afterimage") && !name.equals("Face") && !name.equals("Hair")) {
+                    for (NXNode node : baseNode) {
+                        int itemId = Integer.parseInt(node.getName().replace(".img", ""));
+                        doWork(itemId, node);
+                        loaded.add(itemId);
+                    }
+                }
+             }
+        } catch (IOException e) {
```

Comment:
No. The only other exception that can be thrown after loading is while parsing with integer, short, byte etc. This will throw a NumberFormatException if it goes wrong. I'll add another catch block for this when I get home tonight. Good catch!

Reply:
Done
https://github.com/geospiza-fortis/argonms-server/pull/15/commits/c4dcf7f214d2713266567d5c0cea44722fcd8c7b